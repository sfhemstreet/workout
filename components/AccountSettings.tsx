import styled from "@emotion/styled";
import React, { useState } from "react";
import {
  SwitchTransition,
  Transition,
  TransitionStatus,
} from "react-transition-group";
import { SurfaceElevation } from "../styles/SurfaceElevation";
import { User } from "../types";
import { InlineButton, SecondaryButton, SubmitButton } from "./Buttons";
import { FadeInOutMixin } from "./FadeInOut";
import {
  EMPTY_USERNAME,
  USERNAME_LENGTH,
  USERNAME_TAKEN,
} from "./SignIn/constants";
import { checkUsername } from "./SignIn/helpers";
import { TextInput } from "./TextInput";
import { H1, P } from "./Txt";

type AccountSettingsProps = {
  user: User;

  username: {
    onSubmit: (name: string) => void;
    onTypeName: (name: string) => void;
    isUnique: boolean;
    error: string | undefined;
  };

  avatar: {
    onSubmit: (url: string) => void;
    onCheckUrl: (url: string) => void;
    isUrlOk: boolean;
    error: string | undefined;
  };

  onDeleteAccount: () => void;
  onClose: () => void;
};

/**
 * AccountSettings
 * 
 * Allows signed in users to change their username / avatar, 
 * and for all user's to delete their account.
 * 
 * @param user current User
 * @param username object with properties concerning changing of username
 * @param avatar object with properties concerning changing of avatar
 * @param onDeleteAccount function called after user confirms to delete their account
 * @param onClose function called when user wishes to exit settings
 */
export const AccountSettings = ({
  user,
  username,
  avatar,
  onDeleteAccount,
  onClose,
}: AccountSettingsProps) => {
  const [newUsername, setNewUsername] = useState(user.name);
  const [usernameError, setUsernameError] = useState<string | undefined>(
    undefined
  );
  const [newAvatarUrl, setNewAvatarUrl] = useState(user.avatar);
  const [avatarError, setAvatarError] = useState<string | undefined>(undefined);

  const [isDeletingAccount, setIsDeletingAccount] = useState(false);

  const handleSubmitUsername = () => {
    if (newUsername.trim().length === 0) {
      setUsernameError(EMPTY_USERNAME);
      return;
    }

    if (newUsername.trim() === user.name) {
      setUsernameError("You are already using that name...");
      return;
    }

    if (!username.isUnique) {
      setUsernameError(USERNAME_TAKEN);
      return;
    }

    const { hasLength, hasNoSymbols, hasNoWhitespace } = checkUsername(
      newUsername.trim()
    );

    if (!hasLength) {
      setUsernameError(
        `Name must be ${USERNAME_LENGTH.min}-${USERNAME_LENGTH.max} characters.`
      );
      return;
    }
    if (!hasNoSymbols) {
      setUsernameError(`Name cannot have symbols, besides '-' and '_'`);
      return;
    }
    if (!hasNoWhitespace) {
      setUsernameError(`Name cannot contain whitespace.`);
      return;
    }

    username.onSubmit(newUsername.trim());
  };

  const handleNameChange = (name: string) => {
    setNewUsername(name);
    user.name !== name && username.onTypeName(name.trim());

    if (usernameError) setUsernameError(undefined);
  };

  const handleSubmitAvatar = () => {
    if (newAvatarUrl.trim().length === 0) {
      setAvatarError("Please enter URL to image.");
      return;
    }
    if (!avatar.isUrlOk) {
      setAvatarError("URL must be an image.");
      return;
    }
    if (newAvatarUrl.trim() === user.avatar) {
      setAvatarError("You are already using this as your avatar...");
      return;
    }

    avatar.onSubmit(newAvatarUrl.trim());
  };

  const handleAvatarChange = (url: string) => {
    setNewAvatarUrl(url);
    user.avatar !== url && avatar.onCheckUrl(url);

    if (avatarError) setAvatarError(undefined);
  };

  return (
    <SwitchTransition mode="out-in">
      <Transition
        timeout={{ enter: 0, appear: 100, exit: 300 }}
        key={isDeletingAccount ? "delete" : "edit"}
      >
        {(transitionStatus) =>
          isDeletingAccount ? (
            <Container transitionStatus={transitionStatus}>
              <CloseBtn onClick={onClose}>✕</CloseBtn>
              <H1>Delete Account</H1>
              <P>Are you sure you want to delete your account?</P>
              <Row>
                <SecondaryButton onClick={() => setIsDeletingAccount(false)}>
                  Cancel
                </SecondaryButton>
                <SubmitButton onClick={onDeleteAccount}>Delete</SubmitButton>
              </Row>
            </Container>
          ) : (
            <Container transitionStatus={transitionStatus}>
              <CloseBtn onClick={onClose}>✕</CloseBtn>
              <H1>Account Settings</H1>
              {!user.isAuthenticated && (
                <P>
                  You need to be logged in to change your username or avatar.
                </P>
              )}
              <TextInput
                id="account-settings-name"
                label="Username"
                onChange={handleNameChange}
                placeholder="Want a new username?"
                value={newUsername}
                error={
                  !username.isUnique && newUsername !== user.name
                    ? USERNAME_TAKEN
                    : usernameError
                }
                isOutlined
                isDisabled={!user.isAuthenticated}
              />
              <Row flexEnd>
                <SecondaryButton
                  onClick={handleSubmitUsername}
                  disabled={!user.isAuthenticated}
                >
                  Change Username
                </SecondaryButton>
              </Row>

              <TextInput
                id="account-settings-avatar"
                label="Avatar URL"
                onChange={handleAvatarChange}
                placeholder="Image URl"
                value={newAvatarUrl}
                error={
                  !avatar.isUrlOk
                    ? "Invalid URL, must be an image."
                    : avatarError
                }
                isOutlined
                isDisabled={!user.isAuthenticated}
              />
              <Row flexEnd>
                <SecondaryButton
                  onClick={handleSubmitAvatar}
                  disabled={!user.isAuthenticated}
                >
                  Change Avatar
                </SecondaryButton>
              </Row>
              <br />
              <Row>
                <SecondaryButton onClick={onClose}>Close</SecondaryButton>
                <SecondaryButton onClick={() => setIsDeletingAccount(true)}>
                  Delete Account
                </SecondaryButton>
              </Row>
            </Container>
          )
        }
      </Transition>
    </SwitchTransition>
  );
};

const Container = styled.article<{ transitionStatus: TransitionStatus }>`
  position: relative;
  max-width: 300px;
  width: 100%;
  height: fit-content;

  border-radius: 20px;
  ${(p) => SurfaceElevation(p.theme.name, 3)};

  padding: 1rem;

  ${(p) => FadeInOutMixin(p.transitionStatus)};

  @media ${(p) => p.theme.media.tablet} {
    max-width: 500px;
  }

  button {
    margin-top: 0.25rem;
    margin-bottom: 1rem;
  }
`;

const Row = styled.div<{ flexEnd?: boolean }>`
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: ${(p) => (p.flexEnd ? "flex-end" : "space-around")};
  align-items: center;
`;

const CloseBtn = styled(InlineButton)`
  position: absolute;
  top: 2px;
  right: 12px;
  font-family: ${(p) => p.theme.font.numberFamily};
  font-size: 2rem;
  background: transparent;
`;
