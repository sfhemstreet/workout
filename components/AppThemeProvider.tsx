import { ThemeProvider } from "@emotion/react";
import { ReactNode } from "react";

import { darkColors, lightColors } from "../styles/colors";
import { GlobalStyles } from "../styles/GlobalStyles";
import { theme } from "../styles/theme";
import { ThemeType } from "../types/ThemeType";

type AppThemeProviderProps = {
  children: ReactNode;
  themeType: ThemeType;
};
/**
 * AppThemeProvider
 * - provides the current theme in use to child components.
 * - provides the current theme to Emotion styled components via Emotion's ThemeProvider.
 * - applies global CSS via GlobalStyles.
 *
 * @param children
 */
export const AppThemeProvider: React.FC<AppThemeProviderProps> = ({
  children,
  themeType,
}) => (
  <ThemeProvider
    theme={{
      ...theme,
      name: themeType,
      colors: themeType === "LIGHT" ? lightColors : darkColors,
    }}
  >
    <GlobalStyles themeType={themeType} />
    {children}
  </ThemeProvider>
);
