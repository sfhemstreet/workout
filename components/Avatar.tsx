import styled from "@emotion/styled";
import {
  SwitchTransition,
  Transition,
  TransitionStatus,
} from "react-transition-group";
import { FadeInKeyframe } from "../styles/keyframes";

import { SurfaceElevation } from "../styles/SurfaceElevation";
import { FadeInOutMixin } from "./FadeInOut";
import { LoadingShimmer } from "./LoadingShimmer";


type AvatarProps = {
  isLoading: boolean;
  src: string;
  onClick?: () => void;
};

/**
 * Avatar
 * 
 * Displays round user avatar 
 * 
 * @param isLoading when `true` a LoadingShimmer is rendered instead of avatar
 * @param src src of avatar img
 * @param onClick function to call when avatar is clicked
 */
export const Avatar = ({ isLoading, src, onClick }: AvatarProps) => (
  <BtnContainer
    noHover={onClick === undefined}
    onClick={() => !isLoading && onClick && onClick()}
    aria-label="User information"
  >
    <SwitchTransition mode="out-in">
      <Transition key={isLoading ? "true" : "false"} timeout={200}>
        {(status) => (
          <>
            {isLoading ? (
              <LoadingContainer status={status}>
                <LoadingShimmer />
              </LoadingContainer>
            ) : (
              <Img
                role="button"
                status={status}
                src={src}
                alt="User Avatar"
                title="Toggle user menu"
              />
            )}
          </>
        )}
      </Transition>
    </SwitchTransition>
  </BtnContainer>
);

const BtnContainer = styled.button<{ noHover: boolean }>`
  padding: 0px;
  border: none;
  overflow: hidden;

  ${(p) => SurfaceElevation(p.theme.name, 4)};

  width: 50px;
  height: 50px;
  border-radius: 50%;

  display: flex;
  justify-content: space-around;
  align-items: center;

  animation: ${FadeInKeyframe} 300ms linear both;
  transition: ${(p) => p.theme.transitions.normal};

  cursor: ${(p) => (p.noHover ? "default" : "pointer")};

  :hover {
    ${(p) => !p.noHover && `background: ${p.theme.colors.secondary}`};
  }

  @media ${(p) => p.theme.media.tablet} {
    width: 60px;
    height: 60px;
  }
`;

const Img = styled.img<{ status: TransitionStatus }>`
  width: 46px;
  height: 46px;
  border-radius: 50%;

  ${(p) => FadeInOutMixin(p.status)};

  @media ${(p) => p.theme.media.tablet} {
    width: 56px;
    height: 56px;
  }
`;

const LoadingContainer = styled.div<{ status: TransitionStatus }>`
  width: 46px;
  height: 46px;
  border-radius: 50%;

  overflow: hidden;

  display: flex;
  justify-content: space-around;
  align-items: center;

  transition: ${(p) => p.theme.transitions.normal};
  opacity: ${(p) => (p.status === "entered" ? 1 : 0)};

  @media ${(p) => p.theme.media.tablet} {
    width: 56px;
    height: 56px;
  }
`;
