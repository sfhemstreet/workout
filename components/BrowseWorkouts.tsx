import { css } from "@emotion/react";
import styled from "@emotion/styled";
import React, { useState } from "react";
import { SwitchTransition, Transition } from "react-transition-group";

import {
  useBrowseWorkouts,
  WorkoutsSortType,
} from "../hooks/useBrowseWorkouts";
import { useWorkoutsFilter } from "../hooks/useWorkoutsFilter";
import { ShimmerBackgroundKeyframe } from "../styles/keyframes";
import { SurfaceElevation } from "../styles/SurfaceElevation";
import { FlexSetting } from "../types/FlexSetting";
import { SecondaryButton } from "./Buttons";
import { DifficultySelector } from "./DifficultySelector";
import { Toggler } from "./Toggler";
import { H3, LowEmpSpan } from "./Txt";
import { WorkoutPreview } from "./WorkoutPreview";
import { WorkoutPreviewLoadingShimmers } from "./WorkoutPreviewLoadingShimmers";

export const BrowseWorkouts = React.memo(() => {
  const { filter, onChangeDifficultyFilter } = useWorkoutsFilter();

  const [{ sort, limit }, setOptions] =
    useState<BrowseWorkoutsOptions>(DEFAULT_OPTIONS);

  const { isLoading, workouts, getMoreWorkouts, isExhausted, goToWorkout } =
    useBrowseWorkouts({
      sort,
      limit,
      difficulty: filter.difficulty,
      rating: filter.rating,
    });

  const handleChangeSort = () =>
    setOptions((prevOptions) => ({
      ...prevOptions,
      sort: prevOptions.sort === "LATEST" ? "POPULAR" : "LATEST",
    }));

  const handleChangeLimit = (newLimit: number) =>
    setOptions((prevOptions) => ({ ...prevOptions, limit: newLimit }));

  return (
    <Container>
      <H3>Browse Workouts</H3>
      <Row>
        <Toggler
          onLabel="Latest"
          offLabel="Popular"
          onToggle={handleChangeSort}
          isOn={sort === "LATEST"}
          ariaLabel={`Change sort to ${
            sort === "LATEST" ? "popular" : "latest"
          }`}
          title="Sort"
        />
        <Column>
          <LowEmpSpan padding="1rem 1rem 0rem 1rem">Difficulty</LowEmpSpan>
          <DifficultySelector
            difficulty={filter.difficulty}
            onSelect={onChangeDifficultyFilter}
          />
        </Column>
      </Row>

      <WorkoutsContainer>
        <SwitchTransition mode="out-in">
          <Transition key={isLoading ? "loading" : "results"} timeout={300}>
            {(transitionStatus) =>
              isLoading ? (
                <WorkoutPreviewLoadingShimmers
                  transitionStatus={transitionStatus}
                />
              ) : (
                workouts.map((workout) => (
                  <WorkoutPreview
                    key={`browse-${workout.id}`}
                    workout={workout}
                    onClick={() => goToWorkout(workout.id)}
                  />
                ))
              )
            }
          </Transition>
        </SwitchTransition>
      </WorkoutsContainer>

      {!isExhausted && workouts.length > 0 && (
        <LoadMoreBtn
          disabled={isLoading}
          isLoading={isLoading}
          onClick={getMoreWorkouts}
        >
          More Workouts!
        </LoadMoreBtn>
      )}
    </Container>
  );
});

type BrowseWorkoutsOptions = {
  sort: WorkoutsSortType;
  limit: number;
};

const DEFAULT_OPTIONS: BrowseWorkoutsOptions = {
  sort: "POPULAR",
  limit: 10,
};

const Container = styled.section`
  border-radius: 10px;
  width: 100%;
  height: 100%;
  ${(p) => SurfaceElevation(p.theme.name, 1)};
`;

const WorkoutsContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
`;

const LoadMoreBtn = styled(SecondaryButton)<{
  isLoading: boolean;
}>`
  ${(p) =>
    p.isLoading &&
    css`
      animation: ${ShimmerBackgroundKeyframe} 2s alternate infinite ease-in-out;
      background-image: ${p.theme.colors.superGradient};
      background-size: 400% 100%;
      overflow: hidden;
    `}

  transition: ${(p) => p.theme.transitions.normal};

  margin: 1rem;
`;

export const Row = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row;
  flex-wrap: wrap;
`;

export const Column = styled.div<{
  alignItems?: FlexSetting;
  justifyContent?: FlexSetting;
}>`
  display: flex;
  flex-direction: column;
  justify-content: ${(p) => p.justifyContent ?? "center"};
  align-items: ${(p) => p.alignItems ?? "flex-start"};
`;
