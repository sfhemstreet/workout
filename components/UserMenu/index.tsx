import React from "react";
import { ThemeType, User } from "../../types";
import { SideBar } from "../SideBar";
import { Content } from "./Content";

export type UserMenuProps = {
  isOpen: boolean;
  user: User;
  themeType: ThemeType;
  isSoundOn: boolean;
  actions: {
    signIn: () => void;
    signOut: () => void;
    toggleSettings: () => void;
    toggleTheme: () => void;
    toggleSideBar: () => void;
    toggleSound: () => void;
  };
};

/**
 * UserMenu
 * 
 * Displays User controls.
 * 
 * On small screens Content is rendered inside SideBar component.
 * 
 * @param isOpen controls SideBar  component for small screens
 * @param isSoundOn boolean for sound toggle
 * @param user current User 
 * @param themeType 'DARK' or 'LIGHT'
 * @param actions actions object
 */
export const UserMenu = ({
  isOpen,
  isSoundOn,
  user,
  themeType,
  actions,
}: UserMenuProps) => (
  <>
    <Content
      user={user}
      isSoundOn={isSoundOn}
      themeType={themeType}
      actions={actions}
      isOpen={isOpen}
    />
    <SideBar isOpen={isOpen} side="right" closeSideBar={actions.toggleSideBar}>
      <Content
        user={user}
        isSoundOn={isSoundOn}
        themeType={themeType}
        actions={actions}
        isOpen={isOpen}
        isSideBar={true}
      />
    </SideBar>
  </>
);
