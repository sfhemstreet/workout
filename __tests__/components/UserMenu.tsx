import { UserMenu } from "../../components/UserMenu";
import { render } from "../test-utils/test-util";

test("UserMenu renders", () => {
  const userMenu = render(
    <UserMenu 
      isOpen={true} 
      isSoundOn={true} 
      themeType={"DARK"} 
      actions={{
        signIn: () => {},
        signOut: () => {},
        toggleTheme: () => {},
        toggleSideBar: () => {},
        toggleSound: () => {},
        toggleSettings: () => {}
      }} 
      user={{
        id: "1",
        name: "Test",
        avatar: "",
        isAuthenticated: false,
        error: "",
        isLoading: false,
        workoutsCompleted: []
      }}
    />
  );

  expect(userMenu).toMatchSnapshot();
});
