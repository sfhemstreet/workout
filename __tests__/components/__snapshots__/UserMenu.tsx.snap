// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMenu renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="css-874msf-Container ez6gvys4"
      >
        <div
          class="css-sli15m-AvatarRow ez6gvys2"
        >
          <div
            class="css-xj1fe7-Name ez6gvys3"
          >
            <p
              class="css-1aqmf7z-P-TextMixin elrci8c0"
            >
              Test
            </p>
            <span
              aria-label="Register or sign in to share workouts!"
              data-microtip-position="bottom-left"
              role="tooltip"
            >
              Anonymous
            </span>
          </div>
          <button
            aria-label="User information"
            class="css-1dj8h48-BtnContainer-SurfaceElevation e1odjb1m2"
          >
            <img
              alt="User Avatar"
              class="css-7z81z4-Img-FadeInOutMixin e1odjb1m1"
              role="button"
              src=""
              title="Toggle user menu"
            />
          </button>
        </div>
        <div
          class="css-elxmfa-Menu ez6gvys1"
        >
          <button
            class="css-1tfvr6f-SecondaryButton-SurfaceElevation-SignInButton ez6gvys0"
          >
            Sign In
          </button>
          <p
            class="css-44y8rz-P-TextMixin elrci8c0"
          >
            Workouts Completed:
             
            <span
              class="css-v5jw6b-NumberSpan-TextMixin evnodtx0"
            >
              0
            </span>
             
          </p>
          <span
            class="css-dlk94y-LowEmpSpan-TextMixin ebud7590"
          >
            Settings
          </span>
          <div
            class="css-bqgpbt-Container epl5quv1"
          >
            <span
              class="css-13ikpmi-NumberSpan-TextMixin-Title epl5quv0"
            >
              Theme
            </span>
            <div
              class="css-21c7ls-Row epl5quv7"
            >
              <button
                class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
                disabled=""
              >
                Dark
              </button>
              <div
                class="css-1rkq0zx-ToggleContainer epl5quv3"
              >
                <label
                  aria-label="Change color theme from dark to light."
                  class="css-9qbk92-TogglerLabel epl5quv4"
                >
                  <input
                    checked=""
                    class="css-b547lr-TogglerInput epl5quv5"
                    type="checkbox"
                  />
                  <div
                    class="css-11gan3-ToggleBall epl5quv2"
                  />
                </label>
              </div>
              <button
                class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
              >
                Light
              </button>
            </div>
          </div>
          <div
            class="css-bqgpbt-Container epl5quv1"
          >
            <span
              class="css-13ikpmi-NumberSpan-TextMixin-Title epl5quv0"
            >
              Sound
            </span>
            <div
              class="css-21c7ls-Row epl5quv7"
            >
              <button
                class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
                disabled=""
              >
                ðŸ”Š
              </button>
              <div
                class="css-1rkq0zx-ToggleContainer epl5quv3"
              >
                <label
                  aria-label="Turn off sound."
                  class="css-9qbk92-TogglerLabel epl5quv4"
                >
                  <input
                    checked=""
                    class="css-b547lr-TogglerInput epl5quv5"
                    type="checkbox"
                  />
                  <div
                    class="css-11gan3-ToggleBall epl5quv2"
                  />
                </label>
              </div>
              <button
                class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
              >
                ðŸ”‡
              </button>
            </div>
          </div>
        </div>
      </section>
      <div
        class="css-tko0x8-SideBarBackground-FadeInOutMixin egk50s72"
      >
        <aside
          class="css-1c3lrps-SideBarMenu-SurfaceElevation egk50s71"
        >
          <button
            class="css-wxxhug-OffScreenExitBtn egk50s70"
          >
            Close Side Bar
          </button>
          <section
            class="css-pybnp0-Container ez6gvys4"
          >
            <div
              class="css-sli15m-AvatarRow ez6gvys2"
            >
              <div
                class="css-4djmm-Name ez6gvys3"
              >
                <p
                  class="css-1aqmf7z-P-TextMixin elrci8c0"
                >
                  Test
                </p>
                <span
                  aria-label="Register or sign in to share workouts!"
                  data-microtip-position="bottom-left"
                  role="tooltip"
                >
                  Anonymous
                </span>
              </div>
              <button
                aria-label="User information"
                class="css-1dj8h48-BtnContainer-SurfaceElevation e1odjb1m2"
              >
                <img
                  alt="User Avatar"
                  class="css-7z81z4-Img-FadeInOutMixin e1odjb1m1"
                  role="button"
                  src=""
                  title="Toggle user menu"
                />
              </button>
            </div>
            <div
              class="css-1h152f5-Menu ez6gvys1"
            >
              <button
                class="css-1tfvr6f-SecondaryButton-SurfaceElevation-SignInButton ez6gvys0"
              >
                Sign In
              </button>
              <p
                class="css-44y8rz-P-TextMixin elrci8c0"
              >
                Workouts Completed:
                 
                <span
                  class="css-v5jw6b-NumberSpan-TextMixin evnodtx0"
                >
                  0
                </span>
                 
              </p>
              <span
                class="css-dlk94y-LowEmpSpan-TextMixin ebud7590"
              >
                Settings
              </span>
              <div
                class="css-bqgpbt-Container epl5quv1"
              >
                <span
                  class="css-13ikpmi-NumberSpan-TextMixin-Title epl5quv0"
                >
                  Theme
                </span>
                <div
                  class="css-21c7ls-Row epl5quv7"
                >
                  <button
                    class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
                    disabled=""
                  >
                    Dark
                  </button>
                  <div
                    class="css-1rkq0zx-ToggleContainer epl5quv3"
                  >
                    <label
                      aria-label="Change color theme from dark to light."
                      class="css-9qbk92-TogglerLabel epl5quv4"
                    >
                      <input
                        checked=""
                        class="css-b547lr-TogglerInput epl5quv5"
                        type="checkbox"
                      />
                      <div
                        class="css-11gan3-ToggleBall epl5quv2"
                      />
                    </label>
                  </div>
                  <button
                    class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
                  >
                    Light
                  </button>
                </div>
              </div>
              <div
                class="css-bqgpbt-Container epl5quv1"
              >
                <span
                  class="css-13ikpmi-NumberSpan-TextMixin-Title epl5quv0"
                >
                  Sound
                </span>
                <div
                  class="css-21c7ls-Row epl5quv7"
                >
                  <button
                    class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
                    disabled=""
                  >
                    ðŸ”Š
                  </button>
                  <div
                    class="css-1rkq0zx-ToggleContainer epl5quv3"
                  >
                    <label
                      aria-label="Turn off sound."
                      class="css-9qbk92-TogglerLabel epl5quv4"
                    >
                      <input
                        checked=""
                        class="css-b547lr-TogglerInput epl5quv5"
                        type="checkbox"
                      />
                      <div
                        class="css-11gan3-ToggleBall epl5quv2"
                      />
                    </label>
                  </div>
                  <button
                    class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
                  >
                    ðŸ”‡
                  </button>
                </div>
              </div>
            </div>
          </section>
        </aside>
      </div>
    </div>
  </body>,
  "container": <div>
    <section
      class="css-874msf-Container ez6gvys4"
    >
      <div
        class="css-sli15m-AvatarRow ez6gvys2"
      >
        <div
          class="css-xj1fe7-Name ez6gvys3"
        >
          <p
            class="css-1aqmf7z-P-TextMixin elrci8c0"
          >
            Test
          </p>
          <span
            aria-label="Register or sign in to share workouts!"
            data-microtip-position="bottom-left"
            role="tooltip"
          >
            Anonymous
          </span>
        </div>
        <button
          aria-label="User information"
          class="css-1dj8h48-BtnContainer-SurfaceElevation e1odjb1m2"
        >
          <img
            alt="User Avatar"
            class="css-7z81z4-Img-FadeInOutMixin e1odjb1m1"
            role="button"
            src=""
            title="Toggle user menu"
          />
        </button>
      </div>
      <div
        class="css-elxmfa-Menu ez6gvys1"
      >
        <button
          class="css-1tfvr6f-SecondaryButton-SurfaceElevation-SignInButton ez6gvys0"
        >
          Sign In
        </button>
        <p
          class="css-44y8rz-P-TextMixin elrci8c0"
        >
          Workouts Completed:
           
          <span
            class="css-v5jw6b-NumberSpan-TextMixin evnodtx0"
          >
            0
          </span>
           
        </p>
        <span
          class="css-dlk94y-LowEmpSpan-TextMixin ebud7590"
        >
          Settings
        </span>
        <div
          class="css-bqgpbt-Container epl5quv1"
        >
          <span
            class="css-13ikpmi-NumberSpan-TextMixin-Title epl5quv0"
          >
            Theme
          </span>
          <div
            class="css-21c7ls-Row epl5quv7"
          >
            <button
              class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
              disabled=""
            >
              Dark
            </button>
            <div
              class="css-1rkq0zx-ToggleContainer epl5quv3"
            >
              <label
                aria-label="Change color theme from dark to light."
                class="css-9qbk92-TogglerLabel epl5quv4"
              >
                <input
                  checked=""
                  class="css-b547lr-TogglerInput epl5quv5"
                  type="checkbox"
                />
                <div
                  class="css-11gan3-ToggleBall epl5quv2"
                />
              </label>
            </div>
            <button
              class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
            >
              Light
            </button>
          </div>
        </div>
        <div
          class="css-bqgpbt-Container epl5quv1"
        >
          <span
            class="css-13ikpmi-NumberSpan-TextMixin-Title epl5quv0"
          >
            Sound
          </span>
          <div
            class="css-21c7ls-Row epl5quv7"
          >
            <button
              class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
              disabled=""
            >
              ðŸ”Š
            </button>
            <div
              class="css-1rkq0zx-ToggleContainer epl5quv3"
            >
              <label
                aria-label="Turn off sound."
                class="css-9qbk92-TogglerLabel epl5quv4"
              >
                <input
                  checked=""
                  class="css-b547lr-TogglerInput epl5quv5"
                  type="checkbox"
                />
                <div
                  class="css-11gan3-ToggleBall epl5quv2"
                />
              </label>
            </div>
            <button
              class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
            >
              ðŸ”‡
            </button>
          </div>
        </div>
      </div>
    </section>
    <div
      class="css-tko0x8-SideBarBackground-FadeInOutMixin egk50s72"
    >
      <aside
        class="css-1c3lrps-SideBarMenu-SurfaceElevation egk50s71"
      >
        <button
          class="css-wxxhug-OffScreenExitBtn egk50s70"
        >
          Close Side Bar
        </button>
        <section
          class="css-pybnp0-Container ez6gvys4"
        >
          <div
            class="css-sli15m-AvatarRow ez6gvys2"
          >
            <div
              class="css-4djmm-Name ez6gvys3"
            >
              <p
                class="css-1aqmf7z-P-TextMixin elrci8c0"
              >
                Test
              </p>
              <span
                aria-label="Register or sign in to share workouts!"
                data-microtip-position="bottom-left"
                role="tooltip"
              >
                Anonymous
              </span>
            </div>
            <button
              aria-label="User information"
              class="css-1dj8h48-BtnContainer-SurfaceElevation e1odjb1m2"
            >
              <img
                alt="User Avatar"
                class="css-7z81z4-Img-FadeInOutMixin e1odjb1m1"
                role="button"
                src=""
                title="Toggle user menu"
              />
            </button>
          </div>
          <div
            class="css-1h152f5-Menu ez6gvys1"
          >
            <button
              class="css-1tfvr6f-SecondaryButton-SurfaceElevation-SignInButton ez6gvys0"
            >
              Sign In
            </button>
            <p
              class="css-44y8rz-P-TextMixin elrci8c0"
            >
              Workouts Completed:
               
              <span
                class="css-v5jw6b-NumberSpan-TextMixin evnodtx0"
              >
                0
              </span>
               
            </p>
            <span
              class="css-dlk94y-LowEmpSpan-TextMixin ebud7590"
            >
              Settings
            </span>
            <div
              class="css-bqgpbt-Container epl5quv1"
            >
              <span
                class="css-13ikpmi-NumberSpan-TextMixin-Title epl5quv0"
              >
                Theme
              </span>
              <div
                class="css-21c7ls-Row epl5quv7"
              >
                <button
                  class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
                  disabled=""
                >
                  Dark
                </button>
                <div
                  class="css-1rkq0zx-ToggleContainer epl5quv3"
                >
                  <label
                    aria-label="Change color theme from dark to light."
                    class="css-9qbk92-TogglerLabel epl5quv4"
                  >
                    <input
                      checked=""
                      class="css-b547lr-TogglerInput epl5quv5"
                      type="checkbox"
                    />
                    <div
                      class="css-11gan3-ToggleBall epl5quv2"
                    />
                  </label>
                </div>
                <button
                  class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
                >
                  Light
                </button>
              </div>
            </div>
            <div
              class="css-bqgpbt-Container epl5quv1"
            >
              <span
                class="css-13ikpmi-NumberSpan-TextMixin-Title epl5quv0"
              >
                Sound
              </span>
              <div
                class="css-21c7ls-Row epl5quv7"
              >
                <button
                  class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
                  disabled=""
                >
                  ðŸ”Š
                </button>
                <div
                  class="css-1rkq0zx-ToggleContainer epl5quv3"
                >
                  <label
                    aria-label="Turn off sound."
                    class="css-9qbk92-TogglerLabel epl5quv4"
                  >
                    <input
                      checked=""
                      class="css-b547lr-TogglerInput epl5quv5"
                      type="checkbox"
                    />
                    <div
                      class="css-11gan3-ToggleBall epl5quv2"
                    />
                  </label>
                </div>
                <button
                  class="css-4vnsso-TertiaryButton-SurfaceElevation-Btn epl5quv6"
                >
                  ðŸ”‡
                </button>
              </div>
            </div>
          </div>
        </section>
      </aside>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
